aws/resource_aws_ami.go

```diff
@@ -382,6 +407,18 @@ func resourceAwsAmiRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("sriov_net_support", image.SriovNetSupport)
 	d.Set("virtualization_type", image.VirtualizationType)
 	d.Set("ena_support", image.EnaSupport)
+	d.Set("usage_operation", image.UsageOperation)
+	d.Set("platform_details", image.PlatformDetails)
+	d.Set("hypervisor", image.Hypervisor)
+	d.Set("image_type", image.ImageType)
+	d.Set("owner_id", image.OwnerId)
+	d.Set("public", image.Public)
+	if image.ImageOwnerAlias != nil {
+		d.Set("image_owner_alias", image.ImageOwnerAlias)
+	}
+	if image.Platform != nil {
+		d.Set("platform", image.Platform)
+	}
```

Comment:
`d.Set()` should automatically handle pointer values. Reference: https://github.com/hashicorp/terraform-provider-aws/issues/15893

```suggestion
	d.Set("image_owner_alias", image.ImageOwnerAlias)
	d.Set("platform", image.Platform)
```